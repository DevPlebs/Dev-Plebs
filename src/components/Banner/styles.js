import tw from "tailwind.macro"
import styled from "@emotion/styled"
import bg from "../../images/hills.svg"

export const BannerWrapper = styled.header`
  ${tw`bg-no-repeat bg-top relative w-screen`}
  height: 28em;
  background-image: url(${bg});
`

export const Container = tw.div`flex justify-center h-full flex-col ml-20`

export const Title = styled.h1`
  ${tw`text-white font-bold ml-2`}
  font-size: calc(60px + (120 - 60) * ((100vw - 300px) / (1600 - 300)));
  text-shadow: 2px 2px 2px rgba(0, 0, 0, 0.6);
  font-family: "Rubik", sans-serif;
`

export const Subtitle = styled.h2`
  color: #ffdbb5;
  font-family: "Rubik", sans-serif;
  font-weight: 400;
  margin: -1rem 0 0 0.75rem;
  text-shadow: 1px 1px 1px rgba(30, 30, 30, 0.9);
`
