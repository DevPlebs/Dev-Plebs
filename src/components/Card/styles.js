import styled from "@emotion/styled"
import tw from "tailwind.macro"

// Core Card styles
export const DefaultCard = styled.div`
  ${tw`flex flex-wrap shadow bg-white rounded-lg shadow hover:shadow-lg`}
  width: 100%;
  position: relative;
  font-family: montserrat;
  transition: 0.35s;
  :hover {
    transform: ${props => props.hover && "translateY(-10px)"};
  }
`

export const CardHeader = styled.h4`
  ${tw`py-2 pl-2 pr-6 mb-0 text-white rounded-tl-lg`}
  max-height: 2rem;
  font-family: Rubik;
  background-color: #c04abc;
  clip-path: polygon(0 0, 100% 0%, 85% 100%, 0 100%);
`

export const CardBody = styled.div`
  ${tw`p-4 w-full text-lg flex flex-col`}
  h1, h2, h3, h4, h5 {
    font-family: Rubik;
  }
`

export const CardContainer = styled.div`
  ${tw`relative flex flex-row flex-wrap justify-center md:flex-no-wrap`}
`
export const CardText = styled.p`
  color: #000;
`
export default DefaultCard
