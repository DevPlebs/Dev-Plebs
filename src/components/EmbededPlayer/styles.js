import tw from "tailwind.macro"

export const StyledEmbededPlayer = tw.iframe`border-none`
