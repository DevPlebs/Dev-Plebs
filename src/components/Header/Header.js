import React from "react"
import styled from "@emotion/styled"
import { css } from "@emotion/core"
import PropTypes from "prop-types"

// TODO: This should all be part of a theme
const typeMap = new Map([
  ["primary", "rubik"],
  ["secondary", "montserrat"],
])
const fontMap = new Map([
  ["md", "1.5rem"],
  ["lg", "2.4rem"],
  ["xl", "3rem"],
  ["2xl", "4rem"],
  ["3xl", "5rem"],
])
const styleMap = new Map([
  ["regular", "400"],
  ["light-bold", "600"],
  ["bold", "800"],
])

const dynamicStyle = props =>
  css`
    font-family: ${typeMap.get(props.family)};
    color: ${props.color};
    font-size: ${fontMap.get(props.size)};
    font-weight: ${styleMap.get(props.fontWeight)};
    text-align: ${props.align};
    text-shadow: ${props.shadow && "2px 2px 2px rgba(0, 0, 0, 0.3)"};
  `

const StyledHeader = styled.h1`
  ${dynamicStyle};
`

const Header = ({
  family,
  color,
  fontWeight,
  size,
  align,
  shadow,
  children,
  className,
}) => (
  <StyledHeader
    family={family}
    className={className}
    color={color}
    fontWeight={fontWeight}
    size={size}
    align={align}
    shadow={shadow}
  >
    {children}
  </StyledHeader>
)

Header.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
  family: PropTypes.string,
  color: PropTypes.string,
  fontWeight: PropTypes.string,
  size: PropTypes.string,
  align: PropTypes.string,
  shadow: PropTypes.bool,
}

Header.defaultProps = {
  children: "",
  className: "",
  family: "primary",
  color: "white",
  fontWeight: "regular",
  size: "md",
  align: "left",
  shadow: false,
}

export default Header
