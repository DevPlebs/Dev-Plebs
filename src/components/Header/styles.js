import tw from 'tailwind.macro';

export const HeaderWrapper = tw.header`bg-purple-600 mb-8`
export const Container = tw.div`my-0 mx-auto py-4 px-2 max-w-6xl`
export const Brand = tw.h1`m-0`
export const BrandLink = tw.a`text-white no-underline`