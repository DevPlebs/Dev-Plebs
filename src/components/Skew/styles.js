import styled from "@emotion/styled"

export const SVG = styled.div`
  svg {
    position: absolute;
    bottom: 0;
    width: 100%;
    height: 10vh;
  }
`
