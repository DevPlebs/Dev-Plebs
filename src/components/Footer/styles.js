import styled from "@emotion/styled"

import Header from "../Header"
import tw from "tailwind.macro"

export const StyledFooter = styled.footer`
  ${tw`text-center`}
  background-color: #222535;
  color: #fff;
`
export const StyledText = styled.p`
  ${tw`py-4 text-lg text-center m-0 text-gray-300 max-w-6xl mx-auto px-2 leading-normal md:px-8`}
  font-family: montserrat
`

export const StyledSmallText = styled.p`
  ${tw`py-4 text-xs text-center m-0 text-gray-500 max-w-6xl mx-auto px-2 md:px-8`}
  font-family: montserrat
`
export const IconLink = styled.a`
  ${tw`mx-4 text-5xl text-white hover:text-gray-500 cursor-pointer`}
  transition: .3s;
  svg {
    ${tw`my-6`}
  }
`

export const StyledHeader = styled(Header)`
  ${tw`pt-6 pb-2 m-0 max-w-6xl mx-auto px-2 md:px-8`}
`
