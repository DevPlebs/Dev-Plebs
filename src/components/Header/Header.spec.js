import React from "react"
import { create } from "react-test-renderer"

import Header from "./Header"
import { HeaderWrapper, Container, Brand, BrandLink } from "./styles"

describe("Header", () => {
  const testProps = {
    siteTitle: "test",
  }
  const tree = create(<Header {...testProps} />)

  it("renders correctly", () => {
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("contains <HeaderWrapper />", () => {
    const instance = tree.root
    expect(instance.findAllByType(HeaderWrapper).length).toBe(1)
  })

  it("contains <Container />", () => {
    const instance = tree.root
    expect(instance.findAllByType(Container).length).toBe(1)
  })

  it("contains <Brand />", () => {
    const instance = tree.root
    expect(instance.findAllByType(Container).length).toBe(1)
  })

  it("contains <BrandLink />", () => {
    const instance = tree.root
    expect(instance.findAllByType(Container).length).toBe(1)
  })
})
