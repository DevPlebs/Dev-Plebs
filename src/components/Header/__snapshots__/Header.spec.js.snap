// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.emotion-0 {
  font-family: rubik;
  color: white;
  font-size: 1.5rem;
  font-weight: 400;
  text-align: left;
}

<h1
  className=" emotion-0"
  color="white"
  fontWeight="regular"
  size="md"
>
  
</h1>
`;
